pkg/apis/operator/v1alpha1/types.go

```diff
@@ -82,6 +82,22 @@ type RuntimeCluster struct {
 	Settings *Settings `json:"settings,omitempty"`
 }
 
+// Ingress configures the Ingress specific settings of the runtime cluster.
+type Ingress struct {
+	// Domain specifies the IngressDomain of the cluster pointing to the ingress controller endpoint. It will be used
+	// to construct ingress URLs for system applications running in runtime cluster. Once set this field is immutable.
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
+	// +kubebuilder:validation:Pattern="^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$"
+	Domain string `json:"domain"`
+	// Domains specify the ingress domains of the cluster pointing to the ingress controller endpoint. They will be used
+	// to construct ingress URLs for system applications running in runtime cluster.
+	// +optional
```

Comment:
At least one domain is needed therefore this field should not be optional.

Reply:
As of now one of `domain` and `domains` is required. Therefore, I marked both optional for the time being similar to how @timuthy did it in https://github.com/gardener/gardener/pull/8156. Once we get rid of `domain` in the next release `domains` will become mandatory. Does that work?
WDYT?